src/components/DateTimePicker.js

```diff
@@ -114,13 +114,13 @@ const DatePicker = React.createClass({
     const value = e.target.value.toLowerCase().replace('.', '');
 
     const time = value.split(':');
-    const am = value.includes('am');
-    const pm = value.includes('pm');
+    const am = value.indexOf('am');
+    const pm = value.indexOf('pm');
     const minute = Number(time[1].substring(0, 2));
     let hour = Number(time[0]);
 
-    hour = am && hour === 12 ? 0 : hour;
-    hour = pm && hour !== 12 ? hour + 12 : hour;
+    hour = am >= 0 && hour === 12 ? 0 : hour;
```

Comment:
I think `am` and `pm` should remain booleans, can you move `>= 0` up to the `const` declarations?